src/Swarm/Game/State.hs

```diff
@@ -260,6 +261,32 @@ instance Monoid (Notifications a) where
 
 makeLenses ''Notifications
 
+data SolutionSource
+  = ScenarioSuggested
+  | -- | Includes the SHA1 of the program text
+    PlayerAuthored String
```

Comment:
The `String` type certainly does not suggest a hash and the motivation for it is missing. :raised_eyebrow: 

Why do you need a hash of the original text?

Reply:
Added a newtype wrapper and a bit of explanation.

Admittedly the idea is not fully fleshed out yet, but if we have some kind of public leaderboard, we could publish the "fingerprint" of someone's solution without actually exposing the program text.  Out-of-band, a "judge" could verify that a supplied program text does hash to the leaderboard score.

Or the Sha1 can just be proof to yourself of which file on your local disk you used to achieve that score.